File: math/prime/python/prime.py
# Shunta の自作ライブラリ
# https://github.com/NAVYSHUNTA/atcoder-shunta-library/blob/main/math/prime/python/prime.py

# O(√n): 素数判定
def is_prime(n: int) -> bool:
    if n < 2:
        return False
    if n == 2:
        return True
    if n % 2 == 0:
        return False

    _SKIP_EVEN: int = 2
    div: int = 3
    while div * div <= n:
        if n % div == 0:
            return False
        div += _SKIP_EVEN
    return True

File: math/prime/java/Prime.java
// Shunta の自作ライブラリ
// https://github.com/NAVYSHUNTA/atcoder-shunta-library/blob/main/math/prime/java/Prime.java

// 素数クラス
class Prime {
    // O(√n): 素数判定を行うメソッド
    public static boolean isPrime(long n) {
        if (n < 2L) {
            return false;
        }
        if (n == 2L) {
            return true;
        }
        if (n % 2 == 0) {
            return false;
        }

        final long SKIP_EVEN = 2L;
        long div = 3L;
        while (div * div <= n) {
            if (n % div == 0L) {
                return false;
            }
            div += SKIP_EVEN;
        }
        return true;
    }
}
